### ISSUE TITLE
RBAC - proposal

### ISSUE THREAD

**Issue by verdverm** [sysgears/apollo-universal-starter-kit#556]
I'd like to add RBAC permissioning to the project. Wanted to start a discussion before digging too deep.

**Comment by vlasenko** [comment]
@verdverm Yes, it would be good addition to improve permissions system in the current code.

**Comment by verdverm** [comment]
Created PR https://github.com/sysgears/apollo-universal-starter-kit/pull/561

**Comment by verdverm** [comment]
I'm thinking the way to start integrating this is by first breaking the user directory into `user` and `authentication`/`authn`.  Then to upgrade the authn (with some different drivers/plugins/features ), then add authorization after, maybe with groups/orgs?

The first step is separating the concerns of users and authentication, having users probably just implies authentication  and some form of authorization are enable, something like what we have today for defaults.

**Comment by verdverm** [comment]
I brought both the `auth{authn/authz}` and `entities/{user,group,org,serviceAccount}` subdirectories up to the top level so there are 6 "modules" at the root now.

Does either organization have preference with anyone else? I'm not opposed to grouping them together, but the `entities` and `auth` folder ended up only having an index.js that just pulled the "Features" together.

("") for the current naming discussion happening.
